### ISSUE TITLE
failed to deploy on git push

### ISSUE THREAD

**Issue by bacongobbler** [deis/controller#684]
From someone on slack this morning:

```
[8:06 AM] welch: Question on buildpacks:  I’m running a custom buildpack (for a runnable jar fwiw) and I keep getting the following warning:
remote: 2016/04/29 15:05:39 Error running git receive hook [publishing release (Deis controller endpoint http://10.0.237.129:80/v2/hooks/build: expected status code 200, got 500)]
```

His stack trace, which points to [this code](https://github.com/deis/controller/blob/58217262072343fd76a94fe35606e7deba1453be/rootfs/api/models/build.py#L60):

```
ERROR Internal Server Error: /v2/hooks/build
Traceback (most recent call last):
  File "/usr/lib/python3.5/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3.5/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3.5/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib/python3.5/site-packages/rest_framework/viewsets.py", line 87, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python3.5/site-packages/rest_framework/views.py", line 466, in dispatch
    response = self.handle_exception(exc)
  File "/usr/lib/python3.5/site-packages/rest_framework/views.py", line 463, in dispatch
    response = handler(request, *args, **kwargs)
  File "/app/api/views.py", line 532, in create
    super(BuildHookViewSet, self).create(request, *args, **kwargs)
  File "/app/api/views.py", line 154, in create
    return super(BaseDeisViewSet, self).create(request, *args, **kwargs)
  File "/usr/lib/python3.5/site-packages/rest_framework/mixins.py", line 21, in create
    self.perform_create(serializer)
  File "/app/api/viewsets.py", line 20, in perform_create
    obj = serializer.save(owner=self.request.user)
  File "/usr/lib/python3.5/site-packages/rest_framework/serializers.py", line 191, in save
    self.instance = self.create(validated_data)
  File "/usr/lib/python3.5/site-packages/rest_framework/serializers.py", line 846, in create
    instance = ModelClass.objects.create(**validated_data)
  File "/usr/lib/python3.5/site-packages/django/db/models/manager.py", line 122, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/lib/python3.5/site-packages/django/db/models/query.py", line 401, in create
    obj.save(force_insert=True, using=self.db)
  File "/app/api/models/build.py", line 60, in save
    self.app.scale(self.owner, removed)
  File "/app/api/models/app.py", line 264, in scale
    raise EnvironmentError('No build associated with this release')
```

**Comment by dwelch2344** [comment]
Thanks @bacongobbler. This was with beta2 so I'm attempting to repro with beta3 (or better yet, not repro)

**Comment by dwelch2344** [comment]
Issue might have been resolved on `beta3`. Deploy went out but currently getting `503`'s and `deis logs` are all run together. Taking to slack to figure out what's up...

**Comment by helgi** [comment]
@dwelch2344 Did you figure out if beta3 solved this for you?

**Comment by jchauncey** [comment]
moving this to rc1

**Comment by kmala** [comment]
i think this issue is fixed by https://github.com/deis/controller/pull/711.

**Comment by helgi** [comment]
Closing this for now based on @kmala saying it is fixed in #711 - @dwelch2344 feel free to open this back up if there is still a problem
